app/src/main/java/ch/epfl/culturequest/ui/events/tournaments/TournamentActivity.java

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.culturequest.ui.events.tournaments;
+
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import ch.epfl.culturequest.backend.tournament.apis.TournamentManagerApi;
+import ch.epfl.culturequest.backend.tournament.tournamentobjects.Tournament;
+import ch.epfl.culturequest.databinding.ActivityTournamentBinding;
+import ch.epfl.culturequest.social.Profile;
+import ch.epfl.culturequest.ui.leaderboard.LeaderboardFragment;
+import ch.epfl.culturequest.ui.leaderboard.LeaderboardRecycleViewAdapter;
+import ch.epfl.culturequest.utils.AndroidUtils;
+
+public class TournamentActivity extends AppCompatActivity {
+
+    private ActivityTournamentBinding binding;
+
+    private String uid;
+    private Tournament tournament;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // To make the status bar transparent
+        AndroidUtils.removeStatusBar(getWindow());
+        // fetch the tournament from the intent
+        // tournament = getIntent().getStringExtra("tournament_id");
+        tournament = TournamentManagerApi.getTournamentFromSharedPref();
+
+        if (Profile.getActiveProfile() != null) {
+            uid = Profile.getActiveProfile().getUid();
+        } else {
+            uid = "1234";
+        }
+
+        if (tournament == null ) {
+            throw new RuntimeException("Null argument");
+        }
```

Comment:
It might be more adequate to put a try-catch clause here. In fact, exceptions in important event handlers like `onCreate` or `onResume` directly lead to the whole application crash, so we might want to make it very unlikely to happen, even though, theoretically, this activity shouldn't be created if the tournament hasn't been generated, indeed.